[Bug] [Connector-V2] sink trying to update primary key in distribute sink database

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

I am trying to running seatunnel with standalone mysql database and cluster postgresql sink database.
And it gives following ERROR:

It seems that it is trying to update id's value with id condition. how to prevent updating primary columns?

### SeaTunnel Version

2.3.10

### SeaTunnel Config

```conf
env {
  job.name = "mysql-cdc-test"
  parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 5000
}

source {
  MySQL-CDC {
    server-id = 5652
    username = "system"
    password = "xxxxxxxxx"
    table-names = ["seatunnel.role"]
    base-url = "jdbc:mysql://10.xx.xx.xx:3306"
  }
}

transform {
}

sink {
  jdbc {
    url = "jdbc:postgresql://xxxxxxxxx:11345/tmodel"
    driver = "org.postgresql.Driver"
    user = "xxxxx"
    password = "xxxxxxxx"
    generate_sink_sql = true
    data_save_mode = DROP_DATA
    database = "tmodel"
    table = "data.${table_name}"
    primary_keys = ["${primary_key}"]
    enable_upsert = false
    support_upsert_by_query_primary_key_exist = false
  }
}
```

### Running Command

```shell
`seatunnel.sh --config mysql-cdc-test.conf `
```

### Error Exception

```log
Caused by: java.sql.BatchUpdateException: Batch entry 0 UPDATE "tmodel"."data"."role" SET "id" = ('15'::int4), "type" = ('1'::int4), "role_name" = ('NORMAL_ROLE2'), "description" = ('Normal User'), "create_time" = ('2025-03-24 08:45:55.804+08'), "update_time" = ('2025-05-08 13:31:13.768+08') WHERE "id" = ('15'::int4) was aborted: ERROR: Distributed column "id" can't be updated in current version  Call getNextException to see other errors in the batch.
        at org.postgresql.jdbc.BatchResultHandler.handleCompletion(BatchResultHandler.java:186)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:591)
        at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:889)
        at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:913)
        at org.postgresql.jdbc.PgPreparedStatement.executeBatch(PgPreparedStatement.java:1739)
        at org.apache.seatunnel.shade.com.zaxxer.hikari.pool.ProxyStatement.executeBatch(ProxyStatement.java:127)
        at org.apache.seatunnel.shade.com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeBatch(HikariProxyPreparedStatement.java)
        at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.executor.FieldNamedPreparedStatement.executeBatch(FieldNamedPreparedStatement.java:540)
        at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.executor.InsertOrUpdateBatchStatementExecutor.executeBatch(InsertOrUpdateBatchStatementExecutor.java:108)
        at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.executor.BufferReducedBatchStatementExecutor.executeBatch(BufferReducedBatchStatementExecutor.java:89)
        at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.JdbcOutputFormat.attemptFlush(JdbcOutputFormat.java:172)
        at org.apache.seatunnel.connectors.seatunnel.jdbc.internal.JdbcOutputFormat.flush(JdbcOutputFormat.java:136)
        ... 9 more
Caused by: org.postgresql.util.PSQLException: ERROR: Distributed column "id" can't be updated in current version
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2733)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2420)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:580)
        ... 19 more

        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:220)
        ... 2 more
2025-05-08 13:31:41,309 INFO  [s.c.s.s.c.ClientExecuteCommand] [SeaTunnel-CompletableFuture-Thread-0] - run shutdown hook because get close signal
```

### Zeta or Flink or Spark Version

zeta with seatunnel v2.3.10 

### Java or Scala Version

java version "1.8.0_391"
Java(TM) SE Runtime Environment (build 1.8.0_391-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.391-b13, mixed mode)

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresDialect.java
+++ b/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresDialect.java
@@ -43,9 +43,11 @@ import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 import static org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.psql.PostgresTypeConverter.PG_CHAR;
@@ -159,20 +161,26 @@ public class PostgresDialect implements JdbcDialect {
                 Arrays.stream(uniqueKeyFields)
                         .map(this::quoteIdentifier)
                         .collect(Collectors.joining(", "));
+        final Set<String> uniqueKeyFieldsSet = new HashSet<>(Arrays.asList(uniqueKeyFields));
         String updateClause =
                 Arrays.stream(fieldNames)
+                        .filter(fieldName -> !uniqueKeyFieldsSet.contains(fieldName))
                         .map(
                                 fieldName ->
                                         quoteIdentifier(fieldName)
                                                 + "=EXCLUDED."
                                                 + quoteIdentifier(fieldName))
                         .collect(Collectors.joining(", "));
+        String conflictAction =
+                updateClause.isEmpty()
+                        ? "DO NOTHING"
+                        : String.format("DO UPDATE SET %s", updateClause);
         String upsertSQL =
                 String.format(
-                        "%s ON CONFLICT (%s) DO UPDATE SET %s",
+                        "%s ON CONFLICT (%s) %s",
                         getInsertIntoStatement(database, tableName, fieldNames),
                         uniqueColumns,
-                        updateClause);
+                        conflictAction);
         return Optional.of(upsertSQL);
     }
 

--- a/seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresDialectTest.java
+++ b/seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresDialectTest.java
@@ -1,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.psql;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class PostgresDialectTest {
+
+    @Test
+    void testUpsertStatement() {
+        PostgresDialect dialect = new PostgresDialect();
+        final String database = "seatunnel";
+        final String tableName = "role";
+        final String[] fieldNames = {
+            "id", "type", "role_name", "description", "create_time", "update_time"
+        };
+        final String[] doUpdateKeyFields = {"id"};
+        final String[] doNothingKeyFields = {
+            "id", "type", "role_name", "description", "create_time", "update_time"
+        };
+
+        String doUpdateSql =
+                dialect.getUpsertStatement(database, tableName, fieldNames, doUpdateKeyFields)
+                        .orElseThrow(
+                                () ->
+                                        new AssertionError(
+                                                "Expected doUpdateSql String to be present"));
+        Assertions.assertEquals(
+                doUpdateSql,
+                "INSERT INTO \"seatunnel\".\"role\" (\"id\", \"type\", \"role_name\", \"description\", \"create_time\", \"update_time\") VALUES (:id, :type, :role_name, :description, :create_time, :update_time) ON CONFLICT (\"id\") DO UPDATE SET \"type\"=EXCLUDED.\"type\", \"role_name\"=EXCLUDED.\"role_name\", \"description\"=EXCLUDED.\"description\", \"create_time\"=EXCLUDED.\"create_time\", \"update_time\"=EXCLUDED.\"update_time\"");
+        String doNothingSql =
+                dialect.getUpsertStatement(database, tableName, fieldNames, doNothingKeyFields)
+                        .orElseThrow(
+                                () ->
+                                        new AssertionError(
+                                                "Expected doNothingSql String to be present"));
+        Assertions.assertEquals(
+                doNothingSql,
+                "INSERT INTO \"seatunnel\".\"role\" (\"id\", \"type\", \"role_name\", \"description\", \"create_time\", \"update_time\") VALUES (:id, :type, :role_name, :description, :create_time, :update_time) ON CONFLICT (\"id\", \"type\", \"role_name\", \"description\", \"create_time\", \"update_time\") DO NOTHING");
+    }
+}

